qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/MSDFunctionModel.cpp

```diff
@@ -0,0 +1,520 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MSDFunctionModel.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidQtWidgets/Common/FunctionBrowser/FunctionBrowserUtils.h"
+#include <map>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+using namespace MantidWidgets;
+using namespace Mantid::API;
+
+namespace {
+std::map<MSDFunctionModel::ParamID, QString> g_paramName{
+    {MSDFunctionModel::ParamID::GAUSSIAN_HEIGHT, "Height"},
+    {MSDFunctionModel::ParamID::GAUSSIAN_MSD, "Msd"},
+    {MSDFunctionModel::ParamID::PETERS_HEIGHT, "Height"},
+    {MSDFunctionModel::ParamID::PETERS_MSD, "Msd"},
+    {MSDFunctionModel::ParamID::PETERS_BETA, "Beta"},
+    {MSDFunctionModel::ParamID::YI_HEIGHT, "Height"},
+    {MSDFunctionModel::ParamID::YI_MSD, "Msd"},
+    {MSDFunctionModel::ParamID::YI_SIGMA, "Sigma"}};
+}
+
+MSDFunctionModel::MSDFunctionModel() {}
+
+void MSDFunctionModel::clearData() {
+  m_fitType.clear();
+  m_model.clear();
+}
+
+void MSDFunctionModel::setFunction(IFunction_sptr fun) {
+  clearData();
+  if (!fun)
+    return;
+  if (fun->nFunctions() == 0) {
+    const auto name = fun->name();
+    if (name == "MsdGauss" || name == "MsdPeters" || name == "MsdYi") {
```

Comment:
These names need to be moved into an anon namespace as const strings, since we use them over and over again in this file